src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java

```diff
@@ -74,82 +73,43 @@ public boolean generateAllHDLDescriptions(
-        } else {
-          oneLine.append("_vector( ");
-          if (myWires.get(wire) < 0) {
-            if (!myParametersList.containsKey(myWires.get(wire), attrs)) {
-              // FIXME: hard coded String
-              Reporter.report.addFatalError("Internal Error, Parameter not present in HDL generation, your HDL code will not work!");
-              return contents.clear().get();
-            }
-            oneLine.append("(").append(myParametersList.get(myWires.get(wire), attrs)).append("-1)");
-          } else {
-            oneLine.append((myWires.get(wire) == 0) ? "0" : (myWires.get(wire) - 1));
-          }
-          oneLine.append(" DOWNTO 0 );");
-        }
-        contents.add("   SIGNAL {{1}}", oneLine);
-        oneLine.setLength(0);
+        final var type = getTypeIdentifier(myWires.get(wire), attrs);
+        maxNameLength = Math.max(maxNameLength, wire.length());
+        mySignals.put(wire, type);
       }
-
       for (final var reg : myWires.registerKeySet()) {
-        oneLine.append(reg);
-        while (oneLine.length() < SIGNAL_ALLIGNMENT_SIZE) oneLine.append(" ");
-        oneLine.append(": std_logic");
-        if (myWires.get(reg) == 1) {
-          oneLine.append(";");
-        } else {
-          oneLine.append("_vector( ");
-          if (myWires.get(reg) < 0) {
-            if (!myParametersList.containsKey(myWires.get(reg), attrs)) {
-              // FIXME: hard coded String
-              Reporter.report.addFatalError("Internal Error, Parameter not present in HDL generation, your HDL code will not work!");
-              contents.clear();
-              return contents.get();
-            }
-            oneLine.append("(").append(myParametersList.get(myWires.get(reg), attrs)).append("-1)");
-          } else {
-            if (myWires.get(reg) == 0) {
-              oneLine.append("0");
-            } else {
-              oneLine.append((myWires.get(reg) - 1));
-            }
-          }
-          oneLine.append(" DOWNTO 0 );");
-        }
-        contents.add("   SIGNAL {{1}}", oneLine.toString());
-        oneLine.setLength(0);
+        final var type = getTypeIdentifier(myWires.get(reg), attrs);
+        maxNameLength = Math.max(maxNameLength, reg.length());
+        mySignals.put(reg, type);
       }
-
-      final var typedWires = myTypedWires.getTypedWires();
       for (final var wire : typedWires.keySet()) {
-        oneLine.append(wire);
-        while (oneLine.length() < SIGNAL_ALLIGNMENT_SIZE) oneLine.append(" ");
-        oneLine.append(": ").append(typedWires.get(wire)).append(";");
-        contents.add("   SIGNAL {{1}}", oneLine.toString());
-        oneLine.setLength(0);
+        maxNameLength = Math.max(maxNameLength, wire.length());
+        mySignals.put(wire, typedWires.get(wire));
       }
-      contents.add("")
-          .add("BEGIN")
+      // now we add them
+      if (maxNameLength > 0) contents.addRemarkBlock("Here all used signals are defined");
```

Comment:
What about `All used signals are defined here` instead?

Reply:
Personally, I don't see much value in this comment. It just restates, what the code already expresses.

Reply:
Not quite, as also the I/O components and in-lined components looks like "wiring". I prefer to be verbose here for both debugging purposes, as well as being "transparent" in the generated code

Reply:
fixed